src/control/server/storage_scm_test.go

```diff
@@ -399,22 +315,19 @@ func TestFormatScm(t *testing.T) {
 				GetfsStr:   getFsRetStr,
 			}
 			ss := newMockScmStorage(log, config.ext, nil, []scm.Module{},
-				false, mockPrep, msc)
+				mockPrep, msc)
 			ss.formatted = tt.formatted
 
 			results := ScmMountResults{}
 
-			if tt.inited {
-				// Discovery is run in SCS.Setup() and is not
-				// fatal, continue with expected errors to
-				// format as in normal program execution.
-				if err := ss.Discover(); err != nil {
-					if tt.expErrMsg != "" {
-						ExpectError(t, err, tt.expErrMsg, tt.desc)
-					} else {
-						// unexpected failure
-						t.Fatal(tt.desc + ": " + err.Error())
-					}
+			// Provider scan is not fatal, continue with expected
+			// errors to format as in normal program execution.
+			if _, err := ss.provider.Scan(scm.ScanRequest{Rescan: true}); err != nil {
```

Comment:
Feels weird to test the provider.Scan here.

IMO the provider.Scan calls should be wrapped in another scmStorage method to replace Discover.